### ISSUE TITLE
useConfig must be used within WagmiConfig

### ISSUE THREAD

**Issue by SergiuT** [wagmi-dev/wagmi#3021]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Package Version

1.4.1

### Current Behavior

Package version: 1.4.1

I have a react native expo app and I try to integrate wagmi with RainbowKit. After I completed the installation process and open the app I get this error:

Unhandled Rejection (Error): `useConfig` must be used within `WagmiConfig`.

This is the implementation code:

```
const { chains, publicClient, webSocketPublicClient } = configureChains(
    [
      mainnet,
      polygon
    ],
    [publicProvider()]
)
  
const { connectors } = getDefaultWallets({
    appName: 'test',
    projectId: 'b67c616c2f896b215dff2499d07587ed',
    chains,
})
  
const wagmiConfig = createConfig({
    autoConnect: true,
    connectors,
    publicClient,
    webSocketPublicClient,
})

export default function App() {
    return (
        <WagmiConfig config={wagmiConfig}>
            <RainbowKitProvider chains={chains}>
                <ConnectButton />
            </RainbowKitProvider>
        </WagmiConfig>
    )
}
```

It's worth noted that I have a separate package inside my project where I install the dependencies and make a build with webpack so I can have the libraries loaded already when I use them, but I don't see how this can be an issue, the only thing I do is import them from that package

### Expected Behavior

Should work as expected and load the ConnectButton on the app

**Comment by tmm** [comment]
Thanks for reporting. This issue is temporarily closed due to lack of information. Please provide a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) and reopen this issue.

**Comment by TYRONEMICHAEL** [comment]
@SergiuT did you resolve the issue. I am also getting the same issue. Will post here if there is anything I figure out.

**Comment by TYRONEMICHAEL** [comment]
I nuked my node modules and reinstalled and it worked. Probably a stale version of Wagmi cached somewhere.

**Comment by glitch-txs** [comment]
An MRE on this would be very helpful @TYRONEMICHAEL

**Comment by amoweolubusayo** [comment]
Getting this error now.

**Comment by DKosov** [comment]
same issue

**Comment by Blackglade** [comment]
There's definitely some weird behavior going on here. Experiencing this in NextJS (app router) as well.

Something like this does not work and gives me the same error described as above:

```jsx
// wagmi.js

export default function WAGMI({children}){
  return(
    <WagmiConfig config={config}>
      {children}
    </WagmiConfig>
  )
}

//page.js
import WAGMI from './wagmi.js'

export default function Home(){
  const { isConnected, address } = useAccount()

  return(
    <WAGMI>
      <button>Connect Wallet</button>
    </WAGMI>
  )
}

```

but... if I were to wrap the button in another component and send it to Home like this:

```jsx
//page.js
import WAGMI from './wagmi.js'

function App(){
  const { isConnected, address } = useAccount()
  return(
    <button>Connect Wallet</button>
  )
}

export default function Home(){
  return(
    <WAGMI>
      <App />
    </WAGMI>
  )
}
```

it works! strange...

**Comment by tmm** [comment]
All Wagmi hooks must be used inside `<WagmiConfig />. I the first example you posted, `useAccount` is not inside it.

**Comment by DKosov** [comment]
> All Wagmi hooks must be used inside `<WagmiConfig />. I the first example you posted, `useAccount` is not inside it.

It does not help

**Comment by amoweolubusayo** [comment]
Looks like the error is from RainbowKit. My guess is an update was done on wagmi that the RainbowKit team isn't aware of.  I had to study the node_modules file contents. useProvider is not among the available exports available in wagmi anymore. RainbowKit is still importing and using it to fetch providers

**Comment by antonios-b** [comment]
This might be useful for other users:

Not including WagmiConfig or using Wagmi hooks like useAccount() in the **same component** that contains WagmiConfig will cause this error: "Error: `useConfig` must be used within `WagmiConfig`."

**Comment by atomicDev03** [comment]
I update wagmi, viem and @rainbow-me/rainbowkit to last version and error disappear. updaet this package everywhere

**Comment by dinadeljanin** [comment]
> I update wagmi, viem and @rainbow-me/rainbowkit to last version and error disappear. updaet this package everywhere

Which versions?

**Comment by kdbvier** [comment]
> I update wagmi, viem and @rainbow-me/rainbowkit to last version and error disappear. updaet this package everywhere

could you share the version of each modules - @rainbow-me/rainbowkit, wagmi and viem?

**Comment by rafael-abuawad** [comment]
This issue is telling you that you cannot use `any` hook outside of the WagmiConfig, that means that you **SHOULD** be using the hook on a **CHILD** element of WagmiConfig:

As you can see on the documentation,  There is one file that encloses the `<Profile />` inside of `<WagmiConfig> ... </WagmiConfig>` and **THEN** ont he Profile file the sample uses a hook.

Hopefully this can help you! 😃 

```tsx
import { WagmiConfig, createConfig, mainnet } from 'wagmi'
import { createPublicClient, http } from 'viem'

const config = createConfig({
  autoConnect: true,
  publicClient: createPublicClient({
    chain: mainnet,
    transport: http()
  }),
})

function App() {
  return (
    <WagmiConfig config={config}>
      <Profile />
    </WagmiConfig>
  )
}
```

```tsx
import { useAccount, useConnect, useDisconnect } from 'wagmi'
import { InjectedConnector } from 'wagmi/connectors/injected'

function Profile() {
  const { address, isConnected } = useAccount()
  const { connect } = useConnect({
    connector: new InjectedConnector(),
  })
  const { disconnect } = useDisconnect()

  if (isConnected)
    return (
      <div>
        Connected to {address}
        <button onClick={() => disconnect()}>Disconnect</button>
      </div>
    )
  return <button onClick={() => connect()}>Connect Wallet</button>
}
```

**Comment by shrugs** [comment]
i ran into this error in a correctly configured project because i had some funky import loop going on related to exposing some chains from `viem/chains` in a dependency that my next13 project was importing in order to use in `configureChains`... i have no clue what the actual issue was but it may be helpful to future searchers to look at how they're importing chains and to streamline that as much as possible. in my case i had a dependency called `config` that had two files `chains.ts` and `config.ts` (hah), both exported in a bullet file `index.ts`. `chains.ts` exposed some `viem/chains` and `config.ts` imported them. i actually only needed the type of them in this file, so that fixed my issue.

```ts
// NOTE: importing this value directly makes the nextjs build blow up, idk
import type { SUPPORTED_CHAINS } from "./chains";
```

**Comment by johncastorina** [comment]
This just started happening for me too. Last night, everything was working fine. Today, this error shows up. wagmi version 1.4.7

**Comment by nickbytes** [comment]
Another message for future searchers. Hard for me to pin down exactly what was going on here as well. I figured I had some sort of "funky import loop" similar to what @shrugs was talking about, or possibly a module resolution issue.

I created a minimal reproduction that was as close to the create-wagmi rainbowkit example as possible: https://github.com/nickbytes/useconfig-repro

Interesting thing to note:
- when I use bun, I see the `useConfig` error
- when I use pnpm, I **do not** see the `useConfig` error
